plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/ModulesManager.java

```diff
@@ -619,12 +625,19 @@ public static void buildKeysForRegularEntries(IProgressMonitor monitor, ModulesF
                         continue;
                     }
                 }
-                ModulesKey modulesKey = new ModulesKey(m, f);
 
+                packageM = m;
+                packageF = f;
+                while (packageM != "") {
```

Comment:
You shouldn't compare strings this way (it may work if strings are interned, but we shouldn't count on that -- the proper way is using `.equals()`).

Anyways, in this case, the best pattern here isn't comparing with an empty string, but checking if packageM.isEmpty()